scripts/build_nim.sh

```diff
@@ -64,7 +66,12 @@ nim_needs_rebuilding() {
 		if ! git checkout -q ${NIM_COMMIT} 2>/dev/null; then
 			# Pay the price for a non-default NIM_COMMIT here, by fetching everything.
 			# (This includes upstream branches and tags that might be missing from our fork.)
-			git remote add upstream https://github.com/nim-lang/Nim
+			NEW_REMOTE="upstream"
+			if git remote | grep -q "^upstream$"; then
+				# upstream already exists, call it "extra"
+				NEW_REMOTE="extra"
+			fi
+			git remote add "${NEW_REMOTE}" "${NIM_COMMIT_REPO_URL}"
```

Comment:
what happens here if I change the `NIM_COMMIT_REPO` to a third url? ie I first set it to A then to B?

Reply:
Yeah, sorry, this was not ready yet. :-) We do need to delete the `extra` if it's already set, precisely for the reason you're pointing out. Will wrap this up and ping.

Reply:
OK, I added the delete and did a bit of testing - seems to work as expected. The only potential weirdness I see is that if upstream was not set for some reason and the user does as you said; i.e., sets `NIM_COMMIT_REPO` to A, then to B, then to C, you'll end up with A being set as `upstream` (once), then B set to `extra`, then C set to `extra`, with A sort of perpetually lingering around as upstream.

I'm it would probably make more sense to, in the event of an unknown commit:

* set `upstream` always to https://github.com/nim-lang/Nim/;
* set whatever is supplied in `NIM_COMMIT_REPO`, if anything, to `extra`.

Reply:
OK, that's what I have for now (pushed). :-)